### PR TITLE
tests(card-browser-view-model): sorting & suspend

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
Increase test coverage before we refactor

## How Has This Been Tested?
⚠️ Tests only

## Learning (optional, can help others)
`undoableOp` inside `launch` was problematic

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/browser/CardBrowserViewModelTest.kt [hunk 1]
```diff
@@ -311,7 +321,179 @@ class CardBrowserViewModelTest : JvmTest() {
...
+            assertThat("direction is changed", orderAsc)
+        }
+    }
+
+    /*
+     * Note: suspension behavior has been questioned from a performance perspective and is
+     * subject to change
+     *
+     * Needing to know the 'suspended' status of all cards, makes this O(n).
+     * Anki uses the O(1) approach of using the first selected card
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/browser/CardBrowserViewModelTest.kt:375]: I think using the first selected card would be better both from a performance point of view and a compatibility point of view.

**Review by lukstbit — APPROVED** [review]
I'm fine with the changes.

File: AnkiDroid/src/test/java/com/ichi2/anki/browser/CardBrowserViewModelTest.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/browser/CardBrowserViewModelTest.kt:375]: I suspect I'll be outvoted:

My concern is that selection order is less relevant on Android than it is on Desktop: first in the list, or first tapped?

Additionally, we currently have a few issues with reversing the list, which will be solved once we move to the backend for Card Browser rendering

Once this is resolved after the rewrite, I'm happy for these tests to go (hence the comment in the first place)

Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/anki/browser/CardBrowserViewModelTest.kt:375]: I would vote for compatibility and take a side order of performance

However (most important for this...) that would be (should be?) a separate issue tracked apart from this PR which is just adding coverage and doesn't actually do anything other than comment on the behavior...

**Review by mikehardy — APPROVED** [review]
LGTM - reasonable discussion on a behavior item but it's a side conversation from perspective of "merge this or not" unless I'm missing something obvious, so I'm going to go for it

**Pull request merged**